api/src/main/java/org/openmrs/module/pihcore/apploader/apps/patientregistration/SectionsSierraLeone.java

```diff
@@ -115,6 +115,39 @@ private Question getLocalContactPhoneNumber() {
 
         return q;
     }
+
+    /**
+     * By default, this would generate a display template that removes the Country. In Sierra Leone,
+     * we are capturing addresses from 3 countries, Sierra Leone, Liberia and Guinea, therefore we
+     * need the country displayed
+     * @return Question
+     */
+    @Override
+    public Question getAddressQuestion() {
+        Question q = new Question();
+        q.setId("personAddressQuestion");
+        q.setLegend("registrationapp.patient.address");
+        q.setHeader("registrationapp.patient.address.question");
+
+        Field f = new Field();
+        f.setType("personAddress");
+
+        // If there are address hierarchy levels configured, use the address hierarchy widget, otherwise use the standard address widget
+        List<AddressHierarchyLevel> levels = Context.getService(AddressHierarchyService.class).getAddressHierarchyLevels();
+        if (levels != null && levels.size() > 0) {
+            f.setWidget(getAddressHierarchyWidget(levels, null, true));
```

Comment:
I'm not really sure I follow.  I had to look at the default code and compare to see what you actually changed here.  Would be good to be more specific about how this is different in the comments, or even better, call the super() method and then modify the returned question.
That said, it seems like the change you made is to remove the line above this one that reads:
`q.setDisplayTemplate(getAddressHierarchyDisplayTemplate(levels));`

Isn't the right solution to change the implementation of "getAddressHierarchyDisplayTemplate" to include country?